### PR TITLE
OOQP revival

### PR DESCRIPTION
**Pull request by bnovoselnik**
This PR includes a revived OOQP OCP QP interface and a new OOQP dense QP interface.

### CODE REVIEW

**Comment by bnovoselnik** [comment]
There is some weird linking error on Travis that I'm not getting locally on my laptop:

../../external/OOQP/lib/libooqpdense.a(DeSymIndefSolver.o): In function `DeSymIndefSolver::matrixChanged()':
DeSymIndefSolver.C:(.text+0x5b): undefined reference to `dsytrf_'
../../external/OOQP/lib/libooqpdense.a(DeSymIndefSolver.o): In function `DeSymIndefSolver::solve(OoqpVector&)':
DeSymIndefSolver.C:(.text+0xdb): undefined reference to `dsytrs_'
collect2: error: ld returned 1 exit status

File: examples/c/mass_spring_model/mass_spring_qp.c [hunk 1]
```diff
@@ -212,7 +212,7 @@ ocp_qp_in *create_ocp_qp_in_mass_spring(void *config, ocp_qp_dims *dims)
     int *nu = dims->nu;
     int *nb = dims->nb;
     int *ng = dims->ng;
-    int *ns = dims->ns;
+    // int *ns = dims->ns;
```
Comment (dkouzoup) [inline examples/c/mass_spring_model/mass_spring_qp.c]: this will create a problem if one comments the soft_constraints back in, no?

**Comment by dkouzoup** [comment]
Looks pretty good! Can you add OOQP in most unit tests too?

File: examples/c/mass_spring_model/mass_spring_qp.c (hunk 1 above)
Comment (bnovoselnik) [reply examples/c/mass_spring_model/mass_spring_qp.c]: no, there is a separate function for the case with soft constraints
create_ocp_qp_in_mass_spring_soft_constr

**Comment by bnovoselnik** [comment]
@dkouzoup yep

**Comment by dkouzoup** [comment]
@bnovoselnik you mean already there or working on it?

**Comment by bnovoselnik** [comment]
@dkouzoup already there, but it fails in some cases so I'm trying to figure out is a bug in the interface causing these failures

**Comment by dkouzoup** [comment]
what's up with Travis here?

**Comment by bnovoselnik** [comment]
¯\_(ツ)_/¯ 
no clue, maybe try restarting it manually?

**Comment by dkouzoup** [comment]
Ok! Just did. Is it ready to merge in general or are you still working on bugs/features?

**Comment by bnovoselnik** [comment]
good to go in general, afaic

**Comment by bnovoselnik** [comment]
huh, weird, it seems that unit_tests produces a segfault (when trying to call dense_ooqp) but I'm not getting that locally on my laptop

**Comment by dkouzoup** [comment]
not the first time travis has a different opinion.. try both with release and debug mode maybe?

**Comment by bnovoselnik** [comment]
tried it, both work fine on my end

**Comment by dkouzoup** [comment]
check that you are using one of the travis compilers and try to find on which command the unit tests fail. Because I see that they're also passed at some point.

**Comment by bnovoselnik** [comment]
I just tried with  clang (because I have the exact same version as travis) and it worked again on my laptop.
on travis it fails the first time ooqp is called

**Comment by bnovoselnik** [comment]
@dkouzoup 1e-9 is too extreme for DENSE_HPIPM :grin:

**Pull request merged**